### DISCUSSION TITLE
Running NEON_PRISM configuration when using CESM_LAB in a Docker environment?

### DISCUSSION THREAD

**Discussion by wwieder** [NCAR/CTSM-Tutorial#75] [Q&A]
@tkavoo asked: 

Is it possible to run the NEON_PRISM configuration when using CESM_LAB in a Docker environment? Have tried following the tutorials but does not seem to work at all as it does not recognize the additional command to initiate the PRISM configuration.

Thank you

**Comment by wwieder** [comment]
I think we need to update the CTSM tag in the container. You could try to a git pull on the CTSM repo in the container.

**Comment by wwieder** [comment]
@tkavoo responded:

I tried pulling the CTSM repo in the container. That was successful and when I tried following the notebooks here git clone https://github.com/NCAR/CTSM-Tutorial, especially the customizeCase_PRISM.ipynb gave all errors but I am running the commands on the terminal and it all seems to be working well. 

BUT A follow-up on this one, so I opted to execute the commands in the terminal. But the ./case.build keeps returning an error and never executes:

@[tkavoo](https://github.com/tkavoo)

posted the build log

```
WARNING: buildlib is being called as a program rather than a subroutine as it is expected to be in the CESM context
make complib -j 4 COMP_NAME=clm COMPLIB=/home/user/CLM-NEON/TALL/bld/gnu/mpi-serial/nodebug/nothreads/nuopc/nuopc/esmf/lib/libclm.a -f /home/user/CLM-NEON/TALL.PRISM.transient/Tools/Makefile CIME_MODEL=cesm SMP=FALSE CASEROOT="/home/user/CLM-NEON/TALL.PRISM.transient" CASETOOLS="/home/user/CLM-NEON/TALL.PRISM.transient/Tools" CIMEROOT="/home/user/CTSM/cime" SRCROOT="/home/user/CTSM" COMP_INTERFACE="nuopc" COMPILER="gnu" DEBUG="FALSE" EXEROOT="/home/user/CLM-NEON/TALL/bld" RUNDIR="/home/user/CLM-NEON/TALL.PRISM.transient/run" INCROOT="/home/user/CLM-NEON/TALL/bld/lib/include" LIBROOT="/home/user/CLM-NEON/TALL/bld/lib" MACH="container" MPILIB="mpi-serial" NINST_VALUE="c1a1l1" OS="LINUX" PIO_VERSION=2 SHAREDLIBROOT="/home/user/CLM-NEON/TALL/bld" SMP_PRESENT="FALSE" USE_ESMF_LIB="TRUE" USE_MOAB="FALSE" COMP_ATM="datm" COMP_ICE="sice" COMP_GLC="sglc" COMP_LND="clm" COMP_OCN="socn" COMP_ROF="srof" COMP_WAV="swav" USE_TRILINOS="FALSE" USE_ALBANY="FALSE" USE_PETSC="FALSE" :

output:

err:

make: *** /home/user/CLM-NEON/TALL.PRISM.transient/Tools/Makefile: Resource deadlock avoided. Stop.

ERROR: Command make complib -j 4 COMP_NAME=clm COMPLIB=/home/user/CLM-NEON/TALL/bld/gnu/mpi-serial/nodebug/nothreads/nuopc/nuopc/esmf/lib/libclm.a -f /home/user/CLM-NEON/TALL.PRISM.transient/Tools/Makefile CIME_MODEL=cesm SMP=FALSE CASEROOT="/home/user/CLM-NEON/TALL.PRISM.transient" CASETOOLS="/home/user/CLM-NEON/TALL.PRISM.transient/Tools" CIMEROOT="/home/user/CTSM/cime" SRCROOT="/home/user/CTSM" COMP_INTERFACE="nuopc" COMPILER="gnu" DEBUG="FALSE" EXEROOT="/home/user/CLM-NEON/TALL/bld" RUNDIR="/home/user/CLM-NEON/TALL.PRISM.transient/run" INCROOT="/home/user/CLM-NEON/TALL/bld/lib/include" LIBROOT="/home/user/CLM-NEON/TALL/bld/lib" MACH="container" MPILIB="mpi-serial" NINST_VALUE="c1a1l1" OS="LINUX" PIO_VERSION=2 SHAREDLIBROOT="/home/user/CLM-NEON/TALL/bld" SMP_PRESENT="FALSE" USE_ESMF_LIB="TRUE" USE_MOAB="FALSE" COMP_ATM="datm" COMP_ICE="sice" COMP_GLC="sglc" COMP_LND="clm" COMP_OCN="socn" COMP_ROF="srof" COMP_WAV="swav" USE_TRILINOS="FALSE" USE_ALBANY="FALSE" USE_PETSC="FALSE" failed with rc=2

```

**Reply by wwieder** [reply]
@tkavoo, you may also need to update your externals in the container.  From your ctsm directory try `./manage_externals/checkout_externals`

**Reply by wwieder** [reply]
Note: We're also hoping to update the docker image with an updated CTSM codebase before too long.

**Reply by wwieder** [reply]
After doing this you may need to create a new case for your TALL.PRISM site.

**Reply by tkavoo** [reply]
Thank you for getting back to me. Yes I executed the 
./manage_externals/checkout_externals
I still cannot get the build to finish successfully.
Thank you for the updates and looking forward to the updated CTSM codebase. Also, is there any other alternative to work with the Prism ppt data? Can we download the PRISM-forcing data and use it with a new case?

**Reply by wwieder** [reply]
I honestly haven't tried this in the container.  @TeaganKing or @danicalombardozzi do you have any suggestions for Thomas?
Alternatively, after @briandobbins returns we should update the container and test this functionality?
